components/camel-fastjson/src/main/java/org/apache/camel/component/fastjson/FastjsonDataFormat.java

```diff
@@ -0,0 +1,194 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.fastjson;
+
+import java.io.*;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.alibaba.fastjson.JSON;
+import com.alibaba.fastjson.serializer.SerializerFeature;
+import com.alibaba.fastjson.support.config.FastJsonConfig;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.spi.DataFormat;
+import org.apache.camel.spi.DataFormatName;
+import org.apache.camel.support.ServiceSupport;
+import org.apache.camel.util.IOHelper;
+
+/**
+ * A <a href="http://camel.apache.org/data-format.html">data format</a> ({@link DataFormat})
+ * using <a href="https://github.com/alibaba/fastjson">Fastjson</a> to marshal to and from JSON.
+ */
+public class FastjsonDataFormat extends ServiceSupport implements DataFormat, DataFormatName {
+
+    private FastJsonConfig config;
+    private Class<?> unmarshalType;
+    private Type unmarshalGenericType;
+    private boolean serializeNulls;
+    private boolean prettyPrint;
+    private String dateFormatPattern;
+    private boolean contentTypeHeader = true;
+
+    public FastjsonDataFormat() {
+        this(Object.class);
+    }
+
+    public FastjsonDataFormat(Class<?> unmarshalType) {
+        this(null, unmarshalType);
+    }
+
+    public FastjsonDataFormat(FastJsonConfig config, Class<?> unmarshalType) {
+        this.config = config;
+        this.unmarshalType = unmarshalType;
+    }
+
+    public FastjsonDataFormat(Type unmarshalGenericType) {
+        this(null, unmarshalGenericType);
+    }
+
+    public FastjsonDataFormat(FastJsonConfig config, Type unmarshalGenericType) {
+        this.config = config;
+        this.unmarshalGenericType = unmarshalGenericType;
+    }
+
+    @Override
+    public String getDataFormatName() {
+        return "json-fastjson";
+    }
+
+    @Override
+    public void marshal(final Exchange exchange, final Object graph, final OutputStream stream) throws Exception {
+
+        int len = JSON.writeJSONString(stream,
+                config.getCharset(),
+                graph,
+                config.getSerializeConfig(),
+                config.getSerializeFilters(),
+                config.getDateFormat(),
+                JSON.DEFAULT_GENERATE_FEATURE,
+                config.getSerializerFeatures());
+
+        if (contentTypeHeader) {
```

Comment:
Just a note that exchange is never null, also that getIn or getOut also is never null (they lazy create a new empty message)